### PR TITLE
Add special-resource-operator to the OpenShift CI

### PR DESCRIPTION
**Pull request by ArangoGutierrez**
Add the project special-resource-operator (SRO) to the OpenShift CI

 - https://github.com/openshift-psap/special-resource-operator

Signed-off-by: Carlos Eduardo Arango Gutierrez <carangog@redhat.com>

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
@petr-muller @hongkailiu please have a look, thanks.

**Comment by ArangoGutierrez** [comment]
@petr-muller 

Thanks for the tips, will work on fixing this today and pass the test.

the operator is still not in the openshift organization, is under https://github.com/openshift-psap/special-resource-operator .

**Comment by ArangoGutierrez** [comment]
I get the same error from `correctly-sharded-config` and `generated-config` test even when running the test from a fresh master.

```bash
[eduardo@fedora-ws release]$ podman run --rm -it -v $(pwd)/ci-operator:/ci-operator:z registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest --from-dir /ci-operator/config/ --to-dir /ci-operator/jobs
WARN[0011] Failed to determine info for prow job config  error="file name was not prefixed with \"ci-operator-jobs-\": \"infra-image-mirroring\"" source-file=/ci-operator/jobs/infra-image-mirroring.yaml
WARN[0011] Failed to determine info for prow job config  error="file name was not prefixed with \"ci-operator-jobs-\": \"infra-periodics\"" source-file=/ci-operator/jobs/infra-periodics.yaml
[eduardo@fedora-ws release]$ git log -1
commit d3c9779fab03fa244dd201550dcec02eb06cc2de (HEAD, upstream/master)
Merge: 977a65cb8 0a9cc26ab
Author: OpenShift Merge Robot <openshift-merge-robot@users.noreply.github.com>
Date:   Tue Feb 18 18:01:09 2020 +0100

    Merge pull request #7198 from droslean/private-is
    
    Create 4.x private imagestreams
```

**Comment by ArangoGutierrez** [comment]
/test pj-rehearse

testing again now that the bot has been added to the `openshift-psap` org

**Comment by ArangoGutierrez** [comment]
@petr-muller @hongkailiu PTAL

All `required` tests are successful, rehearsals keep failing claiming permission denied

File: ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+base_images:
+  base:
+    cluster: https://api.ci.openshift.org
```
Comment (petr-muller) [inline ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml]: @hongkailiu are people supposed to start adding these universally now? I know having `cluster` caused problems in the past...

File: core-services/prow/02_config/_plugins.yaml [hunk 2]
```diff
@@ -80,6 +80,14 @@ approve:
...
   - openshift-knative
   require_self_approval: false
+- ignore_review_state: false
+  repos:
+  - openshift-psap/special-resource-operator
+  require_self_approval: true
+- lgtm_acts_as_approve: true
+  repos:
+  - openshift-psap/special-resource-operator
+  require_self_approval: false
```
Comment (petr-muller) [inline core-services/prow/02_config/_plugins.yaml]: Two separate stanzas for a signle repo (with conflicting `require_self_approval` clauses) does not seem right.

Comment (ArangoGutierrez) [reply core-services/prow/02_config/_plugins.yaml]: I took the example from the other repos in this same file.

Comment (petr-muller) [reply core-services/prow/02_config/_plugins.yaml]: No other repo seems to be present in two different stanzas (and if it is, then it's a bug).

File: ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml (hunk 1 above)
Comment (stevekuznetsov) [reply ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml]: This is fine

File: ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    name: "4.4"
```
Comment (stevekuznetsov) [inline ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml]: `master` should be on 4.5

**Comment by ArangoGutierrez** [comment]
@zvonkok @hongkailiu @stevekuznetsov 
After some updates on the Special-resource-operator repo it self, all the test are passing and succeeded. this PR is ready for review :)

File: ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml (hunk 3 above)
Comment (ArangoGutierrez) [reply ci-operator/config/openshift-psap/special-resource-operator/openshift-psap-special-resource-operator-master.yaml]: 728b503

**Pull request merged**